x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/data/ElementType.java

```diff
@@ -86,4 +105,44 @@ public static ElementType fromJava(Class<?> type) {
     public String pascalCaseName() {
         return pascalCaseName;
     }
+
+    private static final Map<String, ElementType> fromLegacyNames;
+
+    static {
+        ElementType[] values = values();
+        fromLegacyNames = Maps.newMapWithExpectedSize(values.length);
+        for (int i = 0; i < values.length; i++) {
+            assert values[i].writableCode == i;
+            fromLegacyNames.put(values[i].legacyWritableName, values[i]);
+        }
+    }
```

Comment:
This is a static final map. Lets keep it immutable.
```suggestion
    private static final Map<String, ElementType> fromLegacyNames = 
        Arrays.stream(ElementType.values()).collect(toUnmodifiableMap(v -> v.legacyWritableName, Function.identity()));
```